### ISSUE TITLE
buildcraft pipes

### ISSUE THREAD

**Issue by pftbest** [Blood-Asp/GT5-Unofficial#12]
after update from 07.07 steam machines stopped accepting steam from buildcraft fluid pipes.
solar boilers work fine (at least they can output steam to the pipe)
gt 08.09, buildcraft 6.4.14, forge 1388.

**Comment by pftbest** [comment]
UPD:
basic steam turbine is also working fine
looks like only steam machines are broken.

**Comment by pftbest** [comment]
i have decompiled original version 07.07 and your release 08.09 using fernflower decompiler, to look for differences (i'm using fernflower, because its hard to compare sources with all that formatting messed up by jd-gui).
and i have found this method, it looks suspicious.
```diff
    public boolean decreaseStoredEnergyUnits(long aEnergy, boolean aIgnoreTooLessEnergy) {
-      return !this.canAccessData()?false:(this.mHasEnoughEnergy = this.decreaseStoredSteam(aEnergy, false) || this.decreaseStoredEU(aEnergy, aIgnoreTooLessEnergy) || aIgnoreTooLessEnergy && this.decreaseStoredSteam(aEnergy, true));
+      return !this.canAccessData()?false:this.decreaseStoredEU(aEnergy, aIgnoreTooLessEnergy);
    }
```
looks like dead code elimination. compiler removed half of the method for some reason.

**Comment by pftbest** [comment]
fixed in 08.11
